http/src/main/java/com/hazelcast/jet/contrib/http/HttpListenerBuilder.java

```diff
@@ -0,0 +1,192 @@
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.contrib.http.impl.HttpListenerSourceContext;
+import com.hazelcast.jet.json.JsonUtil;
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.jet.pipeline.StreamSource;
+
+import javax.annotation.Nonnull;
+import javax.net.ssl.SSLContext;
+import java.util.Objects;
+
+/**
+ * See {@link HttpListenerSources#builder()}.
+ *
+ * @param <T> the type of the pipeline item.
+ */
+public class HttpListenerBuilder<T> {
+
+    /**
+     * Default port for HTTP(s) listener
+     */
+    public static final int DEFAULT_PORT = 5801;
+
+    private static final int PORT_MAX = 0xFFFF;
+
+    private int port = DEFAULT_PORT;
+    private SupplierEx<SSLContext> sslContextFn;
+    private FunctionEx<byte[], T> mapToItemFn;
+    private Class<T> type;
+
+    HttpListenerBuilder() {
+    }
+
+    /**
+     * Set the port which the source binds and listens.
+     * <p>
+     * For example to bind to port `5802`:
+     * <pre>{@code
+     * builder.port(5802)
+     * }</pre>
+     * <p>
+     * Default value is {@link #DEFAULT_PORT} {@code 5801}.
+     *
+     * @param port the port which the source binds and listens.
+     */
+    @Nonnull
+    public HttpListenerBuilder<T> port(int port) {
+        if (port < 0 || port > PORT_MAX) {
+            throw new IllegalArgumentException("Port out of range: " + port + ". Allowed range [0,65535]");
+        }
+        this.port = port;
+        return this;
+    }
+
+    /**
+     * Set the SSL Context function which will be used to initialize underlying
+     * HTTPs listener for secure connections.
+     * <p>
+     * For example:
+     * <pre>{@code
+     * builder.sslContextFn(() -> {
+     *     SSLContext context = SSLContext.getInstance("TLS");
+     *     KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+     *     KeyStore ks = KeyStore.getInstance("JKS");
+     *     char[] password = "123456".toCharArray();
+     *     ks.load(new FileInputStream("the.keystore"), password);
+     *     kmf.init(ks, password);
+     *     TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+     *     KeyStore ts = KeyStore.getInstance("JKS");
+     *     ts.load(new FileInputStream("the.truststore"), password);
+     *     tmf.init(ts);
+     *     context.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+     *     return context;
+     * })
+     * }</pre>
+     * <p>
+     * Default value is {@code null}. By setting this parameter you create a
+     * HTTPs listener source instead of a HTTP listener source.
+     *
+     * @param sslContextFn the function to create {@link SSLContext} which used
+     *                     to initialize underlying HTTPs listener for secure
+     *                     connections.
+     */
+    @Nonnull
+    public HttpListenerBuilder<T> sslContextFn(@Nonnull SupplierEx<SSLContext> sslContextFn) {
```

Comment:
I like this approach with the `SSLContext` supplier provided. What I'm missing is a possibility to require mutual authentication. The current implementation only allows clients to verify server (`Source`) identity, but the HTTP server trusts all clients.

Ref: https://github.com/xnio/xnio/blob/3.8.0.Final/api/src/main/java/org/xnio/Options.java#L154-L159